File: 2529.cpp
#include <bits/stdc++.h>
using namespace std;
int n = 0;
vector<string> ans;
int checker[10];
char a[100] ; //부등호가 들어있는 배열
bool good(char x, char y, char op){
    if(op=='<'){
        if (x>y) return false;
    }
    if(op=='>'){
        if( x<y) return false;
    }
    return true;
}
void recu(int idx, string nums)
{
    if (idx == n + 1)
    {
        ans.push_back(nums);
        return;
    }
    for (int i = 0; i <= 9; i++)
    {
        if (checker[i])
            continue; // i가 이미 사용되었다!
        if (idx == 0 || good(nums[idx-1], i+'0', a[idx-1])  )
        {
            checker[i] = 1; // i를 사용할꺼야
            recu(idx + 1, nums + to_string(i));
            checker[i] = 0; // i 사용 종료
        }
    }
}

File: 2529.py
n = int(input())
l = input().split()
checker = [0] * 10 #사용했는지 확인
min_v = "999"
max_v = "000"
def good(x, y, op):
    if op==">" and x > y: return True
    if op=="<" and x < y: return True
    return False
def recu(idx, nums, min_v, max_v):
    if idx==n+1:
        min_v = min(min_v, nums)
        max_v = max(max_v, nums)
        return min_v, max_v
    for i in range(10):
        if checker[i]: continue #이미 사용되었다면 그냥 지나간다
        if idx==0 or (good(int(nums[-1]), i, l[idx-1] )):
            checker[i] = 1
            min_v, max_v = recu(idx+1, nums+str(i), min_v, max_v)
            checker[i] = 0
    return min_v, max_v
min_v, max_v = recu(0, "",min_v, max_v)
print(max_v, min_v ,sep="\n" )
    
